src/pds_doi_service/core/db/doi_database.py

```diff
@@ -546,14 +546,31 @@ def _form_query_with_wildcards(column_name, search_tokens):
         named_parameters = ",".join([f":{named_param_id}_{i}" for i in range(len(full_tokens))])
         named_parameter_values = {f"{named_param_id}_{i}": full_tokens[i] for i in range(len(full_tokens))}
 
-        # Set up the named parameters for the GLOB portion of the WHERE used
+        # Next, because we use actually use LIKE and not GLOB (for the case-insensitivity),
+        # we need to convert wildcards from Unix style (*,?) to SQLite style (%,_),
+        # but we first need to escape any existing characters reserved by LIKE (& and _)
+        like_chars = ["%", "_"]
+        glob_chars = ["*", "?"]
+
+        for index, wildcard_token in enumerate(wildcard_tokens):
+            for like_char, glob_char in zip(like_chars, glob_chars):
+                # Escape reserved wildcards used by LIKE
+                wildcard_token = wildcard_token.replace(like_char, f"\\{like_char}")
+
+                # Replace wildcards used by GLOB with equivalent for LIKE
+                wildcard_token = wildcard_token.replace(glob_char, like_char)
+
+                # Update the list of wildcards
+                wildcard_tokens[index] = wildcard_token
+
+        # Set up the named parameters for the LIKE portion of the WHERE used
         # find tokens containing wildcards
-        glob_parameters = " OR ".join(
-            [f"{column_name} GLOB :{named_param_id}_glob_{i}" for i in range(len(wildcard_tokens))]
+        like_parameters = " OR ".join(
+            [f"{column_name} LIKE :{named_param_id}_like_{i}" for i in range(len(wildcard_tokens))]
```

Comment:
Is there a potential SQL injection vulnerability here—if `wildcard_tokens` can contain nasty business?

Reply:
I don't think so, since what we're doing here is creating a parameterized query string (i.e. the use of the `:{named_param_id}_like_{i}` placeholders), and then creating a separate mapping for those parameters to the actual wildcard tokens to query for (see lines 572-574).  I believe this is the standard mechanism for preventing SQL injection since the parameterized values are always treated as "data", and never as part of the SQL statement itself.

Reply:
👍